### PR TITLE
recipe2plan: Supporting persistent+queryable `create` handle storage key generation

### PR DESCRIPTION
**Pull request by alxrsngrtn**
## Summary
Primary changes: 
- Adding a `persistentQueryable`-capable storage key creator to the recipe2plan capabilities resolver.
- Increased test coverage
- updated Goldens

Secondary change:  
- Modified CapabilitiesResolver error messaging to make it easier to debug a recipe. 

### Questions for the reviewer
- What protocols do we want to support? `db`, `memdb`, or both? Currently, just `db` is supported. 
- Is `memdb` queryable or persistentQueryable?

### CODE REVIEW

**Comment by piotrswigon** [comment]
I believe `memdb` is not yet a thing, so don't worry about it.

**Comment by alxrsngrtn** [comment]
Will merge after internal presubmit check.

File: src/runtime/capabilities-resolver.ts [hunk 1]
```diff
@@ -65,14 +65,12 @@ export class CapabilitiesResolver {
 
   constructor(public readonly options: CapabilitiesResolverOptions,
               creators?: StorageKeyCreatorInfo[]) {
+    this.creators = CapabilitiesResolver.getDefaultCreators();
     if (creators) {
```
Comment (mmandlis) [inline src/runtime/capabilities-resolver.ts]: the intention of the `creators` was to override the registered creators.
also with this change (still going through my email backlog, so might be wrong) the ts and the kt capabilities resolver implementations are now different.

Comment (alxrsngrtn) [reply src/runtime/capabilities-resolver.ts]: I'll write a PR to fix this. Thanks for looking this over.

**Pull request merged**